### PR TITLE
Cast to touch based on actual table name in cdb_tablemetadata

### PR DESCRIPTION
**Pull request by gfiorav**
This trick allows to touch based on the current `tabname` instead of the `qualified_table_name` in the model. Fixes errors on renaming. 

We're pretty sure it's safe, testing in staging.

Please CR @juanignaciosl 

cc/ @zenitraM

### CODE REVIEW

File: app/models/table.rb [hunk 1]
```diff
@@ -653,6 +653,7 @@ def make_geom_valid
   end
 
   def name=(value)
+    byebug
```
Comment (ethervoid) [inline app/models/table.rb]: Remove the byebug!

**Comment by juanignaciosl** [comment]
Not much to say. Only that moving from table name to oid parameter might cause issues after backups or user movements, since oid is different. @zenitraM might confirm or dismiss this.

Summon @rafatower :eyes: here, since he's knowledge about regclass and oid quirks.

**Comment by zenitraM** [comment]
User mover does update the OIDs to the new ones (https://github.com/CartoDB/cartodb/blob/master/services/user-mover/import_user.rb#L252) after movement so that should not be an issue.

Even though I don't particularly like using them (they can get a bit tricky), they are heavily used by the ghost tables linking code (which syncs oids / table names) so I can't think on a situation where a table can have a different oid without exploding on the ghost tables code

**Comment by gfiorav** [comment]
I've tested basic stuff in staging and all seems good. I've covered:

+ rename table: invalidated
+ rename map: not invalidated
+ change column: invalidated
+ change type of column: invalidated
+ change name of table: invalidated
+ change wizard stuff: not invalidated
+ change privacy: invalidated
+ change description: not invalidated
+ change tags: not invalidated

Any other idea?

**Comment by juanignaciosl** [comment]
- Sync tables
- Assign organization to user? (this moves the tables from one schema to another).

**Comment by zenitraM** [comment]
> - Assign organization to user? (this moves the tables from one schema to another).
This should not change the table OIDs i think but still worth testing

**Comment by gfiorav** [comment]
Both things work in my testing @zenitraM

**Comment by rafatower** [comment]
Sync tables indeed change the OID cause it generates a new table (thus object id) but the syncing code takes care of that. User mover should be a similar case.

LGTM :+1: especially since it's tested.

**Comment by gfiorav** [comment]
Thanks for looking into it on such a short notice!

It looks like support has a fix for the issue, so I'm :chicken: out and deploying this tomorrow morning.

**Comment by gfiorav** [comment]
Forgot to add the change to NEWS. Deploying once CI passes.

**Pull request merged**